### ISSUE TITLE
Observed regression with cargo after updating from 1.27.0 to 1.28.0

### ISSUE THREAD

**Issue by zazaho** [c-ares/c-ares#737]
I observe persistent download failures with `cargo build`  after upgrading c-ares. Downgrading resolves the issue.

My cargo is built to use system libcurl version 8.7.1

The message I see is:
```
Updating crates.io index
warning: spurious network error (3 tries remaining): [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
warning: spurious network error (2 tries remaining): [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
warning: spurious network error (1 tries remaining): [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
```

**Comment by bradh352** [comment]
I'd need more information probably, such as if this is observed on just one system or multiple systems.  Also, what OS is this on?  My best guess is if you're on Linux, that maybe there's something up with the new `/etc/resolv.conf` and `/etc/nsswitch.conf` parsers so data out of those files might prove useful.

**Comment by zazaho** [comment]
I am sorry for leaving out such detais. I manage only my personal machine which is a fedora based machine but I try to keep it up to date by managing a number of rpm packages myself. As vanilla as possible.

I attach the files you mentioned.
[nsswitch.conf.txt](https://github.com/c-ares/c-ares/files/14812147/nsswitch.conf.txt)
[resolv.conf.txt](https://github.com/c-ares/c-ares/files/14812148/resolv.conf.txt)

**Comment by zazaho** [comment]
Are there (easy) steps that I can take to help identify the origin of issue ?

**Comment by bradh352** [comment]
Nothing looks unusual there.  I'm not familiar with how fedora packages c-ares, do they package the 'ahost' and 'adig' utilities that typically get built with c-ares?  It would be useful to see the output of using those utilities to resolve the same domain name.

**Comment by zazaho** [comment]
[adig_v1.27.0.txt](https://github.com/c-ares/c-ares/files/14812263/adig_v1.27.0.txt)
[adig_v1.28.0.txt](https://github.com/c-ares/c-ares/files/14812264/adig_v1.28.0.txt)
[ahost_v1.27.0.txt](https://github.com/c-ares/c-ares/files/14812266/ahost_v1.27.0.txt)

ahost  of 1.28.0 produces no output but say `crates.io: Domain name not found`

**Comment by bradh352** [comment]
very strange, i've been able to reproduce, its something with searching and the fact that no search domains are configured.
if you add a line like:
```
search doesnotexist.local
```
to your `/etc/resolv.conf`, it will magically start working.

**Comment by bradh352** [comment]
This should be fixed in b00b0ac as it works on my test linux system with the patch and did not without.

**Comment by bradh352** [comment]
and released as part of c-ares 1.28.1 now
